mlflow/store/artifact/s3_artifact_repo.py

```diff
@@ -185,6 +185,11 @@ def log_artifacts(self, local_dir, artifact_path=None):
                 rel_path = os.path.relpath(root, local_dir)
                 rel_path = relative_path_to_artifact_path(rel_path)
                 upload_path = posixpath.join(dest_path, rel_path)
+
+            if not filenames:
+                # We're in an empty directory. Create a folder to preserve the directory structure.
+                s3_client.put_object(Bucket=bucket, Key=upload_path + "/")
```

Comment:
TODO: add try-catch here

Reply:
@B-Step62 I tested this line using my personal s3. It does nothing if the directory already exists.